emodpy_typhoid/utility/sweeping.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+from typing import Dict, Any, List
+from idmtools.entities.simulation import Simulation
+from logging import getLogger, DEBUG
+
+logger = getLogger()
+
+
+##################################################
+# Sweeping utility functions
+##################################################
+def set_param(simulation: Simulation, param: str, value: Any) -> Dict[str, Any]:
+    """
+    Set specific parameter value.
+    Args:
+        simulation: idmtools Simulation
+        param: parameter
+        value: new value
+
+    Returns:
+        dict
+    """
+    # return simulation.task.set_parameter(param, value)
+
+    try:
+        return simulation.task.set_parameter(param, value)
+    except ValueError:
+        if "parameters" in simulation.task.config:
+            config = simulation.task.config.parameters
+        else:
+            config = simulation.task.config
+
+        config[param] = value
+        return {param: value}
+
+
+def sweep_functions(simulation: Simulation, func_list: List) -> Dict[str, Any]:
+    """
+    Apply funcs on simulation.
+    Args:
+        simulation: idmtools Simulation
+        func_list: a list of functions
+
+    Returns:
+        dict of parameters
+    """
+    tags_updated = {}
+    for func in func_list:
+        tags = func(simulation)
+        if tags:
+            tags_updated.update(tags)
+    return tags_updated
+
+
+class ItvFn:
+    """
+    Sweeping utility: works for sweeping on interventions.
+    Requirements:
+     - func is a method that takes campaign as first parameter
+     - func return a dict
+
+    Returns:
+        dict
+    """
+
+    def __init__(self, func, *args, **kwargs):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+    def __call__(self, simulation: Simulation):
+        import emod_api.campaign as campaign
+        campaign.reset()
+
+        md = self.func(campaign, *self.args, **self.kwargs)
+
+        # add events
+        events = campaign.campaign_dict["Events"]
+        simulation.task.campaign.add_events(events)
+
+        # update config for adhoc events
```

Comment:
This should really be handled by emod_api.campaign, and we should be able to get rid of when we get onto a more modern Typhoid_SIM branch, but leave it for now, it's fine.

Reply:
I was wondering this part too. almost wanted to put just pass to if/else.